internal/db/mock.go

```diff
@@ -0,0 +1,390 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+var (
+	_               Querier = &QuerierProxier{}
+	_               driver.Driver  = &DBProxy{}
+	_               driver.Conn    = &ConnProxy{}
+)
+
+type QuerierProxier struct {
+	*Querier
+	OverwrittenHasGrant                                func(ctx context.Context, arg HasGrantParams) (int32, error)
```

Comment:
@jules this doesn't work because HasGrantParams doesn't exist, please use the interface Querier to determine what to put here don't just make it up.